File: app_go/metricas/metricas.go
package metricas

import (
    "github.com/prometheus/client_golang/prometheus"
    "github.com/prometheus/client_golang/prometheus/promauto"
)

var (
    RequisocoesTotal = promauto.NewCounterVec(
        prometheus.CounterOpts{
            Name: "app_requisicoes_total",
            Help: "Total de requisições",
        },
        []string{"metodo", "endpoint", "status"},
    )

    LatenciaRequisicoes = promauto.NewHistogramVec(
        prometheus.HistogramOpts{
            Name:    "app_latencia_requisicoes_segundos",
            Help:    "Latência das requisições em segundos",
            Buckets: prometheus.DefBuckets,
        },
        []string{"endpoint"},
    )

    CacheHits = promauto.NewCounter(
        prometheus.CounterOpts{
            Name: "app_cache_hits_total",
            Help: "Total de cache hits",
        },
    )

    CacheMisses = promauto.NewCounter(
        prometheus.CounterOpts{
            Name: "app_cache_misses_total",
            Help: "Total de cache misses",
        },
    )
)

File: app_python/src/rotas/metricas.py
from prometheus_client import Counter, Histogram, Gauge, generate_latest, CONTENT_TYPE_LATEST
from fastapi import APIRouter, Response

router = APIRouter()

requisicoes_total = Counter(
    'app_requisicoes_total',
    'Total de requisições',
    ['metodo', 'endpoint', 'status']
)

latencia_requisicoes = Histogram(
    'app_latencia_requisicoes_segundos',
    'Latência das requisições em segundos',
    ['endpoint']
)

cache_hits = Counter('app_cache_hits_total', 'Total de cache hits')
cache_misses = Counter('app_cache_misses_total', 'Total de cache misses')

@router.get("/metricas")
def metricas():
    return Response(generate_latest(), media_type=CONTENT_TYPE_LATEST)
